channeld/channel.c

```diff
@@ -487,41 +486,41 @@ static void maybe_send_shutdown(struct peer *peer)
 	peer->shutdown_sent[LOCAL] = true;
 }
 
-/* Master has acknowledged that we're sending commitment, so send it. */
-static void handle_sending_commitsig_reply(struct peer *peer, const u8 *msg)
+/* This queues other traffic from the master until we get reply. */
+static u8 *master_wait_sync_reply(const tal_t *ctx,
+				  struct peer *peer, const u8 *msg,
+				  enum channel_wire_type replytype)
 {
-	status_trace("Sending commit_sig with %zu htlc sigs",
-		     tal_count(peer->next_commit_sigs->htlc_sigs));
+	u8 *reply;
 
-	peer->next_index[REMOTE]++;
+	status_trace("Sending master %s",
+		     channel_wire_type_name(fromwire_peektype(msg)));
 
-	msg = towire_commitment_signed(peer, &peer->channel_id,
-				       &peer->next_commit_sigs->commit_sig,
-				       peer->next_commit_sigs->htlc_sigs);
-	msg_enqueue(&peer->peer_out, take(msg));
-	peer->next_commit_sigs = tal_free(peer->next_commit_sigs);
+	if (!wire_sync_write(MASTER_FD, msg))
+		status_failed(WIRE_CHANNEL_INTERNAL_ERROR,
+			      "Could not set sync write to master: %s",
+			      strerror(errno));
 
-	maybe_send_shutdown(peer);
+	status_trace("... , awaiting %s",
+		     channel_wire_type_name(replytype));
 
-	/* Timer now considered expired, you can add a new one. */
-	peer->commit_timer = NULL;
-	start_commit_timer(peer);
-
-	if (shutdown_complete(peer))
-		io_break(peer);
-}
-
-/* This blocks other traffic from the master until we get reply. */
-static void master_sync_reply(struct peer *peer, const u8 *msg,
-			      enum channel_wire_type replytype,
-			      void (*handle)(struct peer *peer, const u8 *msg))
-{
-	assert(!peer->handle_master_reply);
+	for (;;) {
+		reply = wire_sync_read(ctx, MASTER_FD);
+		if (!reply)
+			status_failed(WIRE_CHANNEL_INTERNAL_ERROR,
+				      "Could not set sync read from master: %s",
+				      strerror(errno));
+		if (fromwire_peektype(reply) == replytype) {
```

Comment:
I'd prefer having some more descriptive message in here, because these may potentially be very far apart from each other.

Reply:
Well, that status will almost certainly never be seen; it can only be caused by the closing of the master fd.